tests/tools/oecert/host/host.cpp

```diff
@@ -196,9 +202,41 @@ static void _display_help(const char* cmd)
     printf("\t%s\n", INPUT_PARAM_OPTION_VERBOSE);
 }
 
+// Get full path of oecert running executable, then get enclave filename by:
+// In linux, replace "<path>/oecert" with "<path>/oecert_enc"
+// In windows, replace "<path>/oecert.exe" with "<path>/oecert_enc"
+static int _get_enclave_filename()
+{
+    char path[OE_PATH_MAX];
+    size_t path_size = 0;
+    char* enclave_filename = nullptr;
+
+#if defined(__linux__)
+    path_size += (size_t)readlink("/proc/self/exe", path, OE_PATH_MAX);
+#elif defined(_WIN32)
+    path_size += (size_t)GetModuleFileName(NULL, path, OE_PATH_MAX);
+    path_size -= strlen(".exe");
+#endif
+
+    if (path_size < 0)
+    {
+        printf("Failed to read enclave path\n");
+        return 1;
+    }
+    path[path_size] = '\0';
+
+    enclave_filename =
```

Comment:
This allocation is never freed.

Since _parse_args() is the only function calling this, you can return allocated string and have caller free the string. I.e. `char* _get_enclave_filename();`

Reply:
Used a global `enclave_filename` like `_params`. Added `free()` at the end of `main`.